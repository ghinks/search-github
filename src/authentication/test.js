import { getEnvToken, getConfigToken, __RewireAPI__ as authRewire } from './index'

describe('authentication', () => {
  let envToken
  const gitconfig = {
    'user': {
      'name': 'glenn vincent hinks',
      'email': 'ghinks@yahoo.com'
    },
    'core': {
      'excludesfile': '/Users/developer/.gitignore_global',
      'autocrlf': 'input'
    },

    'github-search': {
      'token': 'abc'
    }
  }

  beforeAll(() => {
    envToken = process.env.GITHUB_TOKEN
    process.env.GITHUB_TOKEN = 'abc'
    authRewire.__Rewire__('parse', { sync: () => gitconfig })
  })
  afterAll(() => {
    process.env.GITHUB_TOKEN = envToken
    authRewire.__ResetDependency__('parse')
  })
  test('get github token', () => {
    expect(getEnvToken()).toMatchSnapshot()
  })

  test('get config token', () => {
    expect(getConfigToken()).toMatchSnapshot()
  })
})
