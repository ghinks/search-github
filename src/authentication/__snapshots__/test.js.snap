// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authentication get config token 1`] = `"abc"`;

exports[`authentication get github token 1`] = `"abc"`;
